TV/GameModes/BriansGames/BattleReady/BattleReadyGameMode.cs

```diff
@@ -0,0 +1,251 @@
+using static System.FormattableString;
+using RoystonGame.TV.DataModels;
+using RoystonGame.TV.GameModes.Common.ThreePartPeople.DataModels;
+using System;
+using System.Linq;
+using Connector = System.Action<
+    RoystonGame.TV.DataModels.User,
+    RoystonGame.TV.DataModels.Enums.UserStateResult,
+    RoystonGame.Web.DataModels.Requests.UserFormSubmission>;
+
+namespace RoystonGame.TV.GameModes.BriansGames.BattleReady
+{
+    public class BattleReadyGameMode : IGameMode
+    {
+        private List<PeopleUserDrawing> Drawings { get; set; } = new List<PeopleUserDrawing>();
+        private List<Prompt> Prompts { get; set; } = new List<Prompt>();
+        private GameState Setup { get; set; }
+        private List<GameState> Gameplays { get; set; } = new List<GameState>();
+        private List<GameState> Scoreboards { get; set; } = new List<GameState>();
+        private List<GameState> DisplayPeoples { get; set; } = new List<GameState>();
+        private Lobby gameLobby { get; set; }
+        private RoundTracker RoundTracker = new RoundTracker();
+        private Random Rand { get; set; } = new Random();
+        public BattleReadyGameMode(Lobby lobby, List<ConfigureLobbyRequest.GameModeOptionRequest> gameModeOptions)
+        {
+            ValidateOptions(gameModeOptions);
+
+            List<(User, string)> promptTuples = new List<(User, string)>();
+            List<Prompt> promptsCopy = new List<Prompt>();
+            int numRounds = int.Parse(gameModeOptions[0].ShortAnswer);
+            int numPromptForEachUsersPerRound = int.Parse(gameModeOptions[1].ShortAnswer);
+            int numDrawingsPerPersonPerPart = int.Parse(gameModeOptions[2].ShortAnswer);
+            int numOfEachPartInHand = 3;
+            int numUsersPerPrompt = 2;
+
+            int numPromptsEachRound = numPromptForEachUsersPerRound * lobby.GetActiveUsers().Count / numUsersPerPrompt;
+            int numPromptsNeededFromUser = numRounds * numPromptForEachUsersPerRound / numUsersPerPrompt;
+
+            Setup = new Setup_GS(
+                lobby: lobby, 
+                drawings: Drawings,
+                prompts: promptTuples,
+                numDrawingsPerUserPerPart: numDrawingsPerPersonPerPart,
+                numPromptsPerUser: numPromptsNeededFromUser);
+
+            Setup.AddStateEndingListener(() =>
+            {
+                List<PeopleUserDrawing> randomizedHeads = Drawings.FindAll((drawing) => drawing.Type == DrawingType.Head).OrderBy(_ => Rand.Next()).ToList();
+                List<PeopleUserDrawing> randomizedBodies = Drawings.FindAll((drawing) => drawing.Type == DrawingType.Body).OrderBy(_ => Rand.Next()).ToList();
+                List<PeopleUserDrawing> randomizedLegs = Drawings.FindAll((drawing) => drawing.Type == DrawingType.Legs).OrderBy(_ => Rand.Next()).ToList();
+                while (Drawings.Count < lobby.GetActiveUsers().Count* numPromptForEachUsersPerRound* 3*numOfEachPartInHand)
```

Comment:
This makes it clearer that the while loop cant go on forever
```suggestion
                int totalDrawings = lobby.GetActiveUsers().Count* numPromptForEachUsersPerRound* 3*numOfEachPartInHand;
                while (Drawings.Count < totalDrawings)
```